contracts/tracer/src/msg.rs

```diff
@@ -0,0 +1,38 @@
+use cosmwasm_std::Binary;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::state::LogEntry;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    Touch {},
+    Fail {},
+    Forward {
+        addr: String,
+        msg: Binary,
+        #[serde(default)]
+        marker: u64,
+        #[serde(default)]
+        catch_success: bool,
+        #[serde(default)]
+        catch_failure: bool,
+        #[serde(default)]
+        fail_reply: bool,
+    },
+    Clear {},
+    Reset {},
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    Log { depth: Option<u32> },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct LogResponse {
+    pub log: Vec<Vec<LogEntry>>,
```

Comment:
Because we want to have a view on change history. The sole purpose of this contract is to show how things changed step by step on more complicated sub calls sequences.

In other words - every element of a vector is a whole log from the beginning. Technically it is obvious for us, that it would just be extended one by one, but the purpose is to not assume we know it happens, and there are no strange ordering things, it is to show it using this contract.